### ISSUE TITLE
Version 2.13.0 Get-MGUser and Filter throws an error

### ISSUE THREAD

**Issue by fug4zi** [microsoftgraph/msgraph-sdk-powershell#2575]
With the update to Version 2.13.0 the user filter throws an exception.

```
Get-MgUser -All -Filter "assignedLicenses/`$count ne 0" -ConsistencyLevel eventual -CountVariable UserCount
Get-MgUser : Object reference not set to an instance of an object.
At line:1 char:1
+ Get-MgUser -All -Filter "assignedLicenses/`$count ne 0" -ConsistencyL ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgUser_List], NullReferenceException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_List
```

Same here:
```
$user = "xxx@xxx.de"
Get-MgUser -Filter "Mail eq '$($user)' or otherMails/any(id:id eq '$($user)')"
Get-MgUser : Object reference not set to an instance of an object.
At line:1 char:70
+ ... r -Filter "Mail eq '$($user)' or otherMails/any(id:id eq '$($user)')"
+                                                                 ~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-MgUser_List], NullReferenceException
    + FullyQualifiedErrorId : Microsoft.Graph.PowerShell.Cmdlets.GetMgUser_List
```

**Comment by Skorfulose** [comment]
Same for me, with 2.13.0 and multiple cmdlets. Downgraded to 2.12.0 and it's gone again.

**Comment by xyanide** [comment]
Same for Get-MgServicePrincipalAppRoleAssignment. Works fine with a previous version

![image](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/1845578/ad3c01b7-1287-45ca-882c-cba25f9094f8)

**Comment by CatalinTRSK** [comment]
Same for Get-MgServicePrincipal_List: Object reference not set to an instance of an object.
And probably, lot of other cmdlets are break. And that is not the first 2x release that breaks everything  in production ...
Shall we expect a bug solving soon? Or shall we stick with 2.12 until next major release?

**Comment by PoshlandPro** [comment]
I have the same. I needed to downgrade module to 2.12.0 on my Azure Function instance as a workaround.

**Comment by aniket-sinha1** [comment]
Same issue.
Can we expect this to get fixed asap?

**Comment by dantstore** [comment]
Same issue here for Get-MgApplication, which has broken our build pipelines.
Will be downgrading to 2.12.0.

**Comment by timayabi2020** [comment]
Hi all. Apologies for this. The fix is available on 2.13.1.

**Comment by dantstore** [comment]
Can confirm that the fix has worked for me.

**Comment by amorrowbellarmine** [comment]
@timayabi2020 Thank you for the fix, but now 2.13.1 is causing other issues by returning "ResponseHeaders" in the output of several functions, like Get-MgUser. Please see issues https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/2581 and https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/2584 .

**Comment by justinmontgomeryCF** [comment]
2.13.1 is not working for me.

Get-MGUser still returns ResponseHeaders, as does Update-MgDirectoryOnPremiseSynchronization
